### PR TITLE
WEB 814 - Redirect to My NFTs once bought

### PR DESCRIPTION
**Pull request by joesobo**

### CODE REVIEW

File: components/BuyAssetForm/index.tsx [hunk 1]
```diff
@@ -52,12 +52,13 @@ const BuyAssetForm = ({
     }
 
     try {
+      const chainAccount = currentUser.actor;
       const purchaseResult = await ProtonSDK.purchaseSale({
-        buyer: currentUser ? currentUser.actor : '',
+        buyer: currentUser ? chainAccount : '',
```
Comment (joannerd) [inline components/BuyAssetForm/index.tsx:57]: Let's keep it as it was so we don't possibly receive an `unable to access "actor" or undefined object` sort of error.

File: components/BuyAssetForm/index.tsx [hunk 2]
```diff
@@ -52,12 +52,13 @@ const BuyAssetForm = ({
...
         sale_id: saleId,
       });
       if (purchaseResult.success) {
-        router.replace(router.asPath);
+        router.push(`/my-nfts/${chainAccount}`);
```
Comment (joannerd) [inline components/BuyAssetForm/index.tsx:61]: I remember we decided to push to the asset detail page `/assets/???` instead.

File: components/BuyAssetForm/index.tsx (hunk 1 above)
Comment (joannerd) [reply components/BuyAssetForm/index.tsx:57]: Scratch that! We should just set an error and return early when `currentUser` is undefined.

File: components/BuyAssetForm/index.tsx [hunk 3]
```diff
@@ -52,12 +52,17 @@ const BuyAssetForm = ({
     }
 
     try {
+      if (!currentUser) {
```
Comment (cindyssun5) [inline components/BuyAssetForm/index.tsx:55]: Best to also check if there's currentUser.actor: `if (!currentUser || !currentUser.actor)`

**Review by cindyssun5 — CHANGES_REQUESTED** [review]

**Review by cindyssun5 — APPROVED** [review]

**Pull request merged**